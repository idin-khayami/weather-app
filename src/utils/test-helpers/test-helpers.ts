import { ReactElement } from 'react';

import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { UserEvent, userEvent } from '@testing-library/user-event';

export const renderWithUserSetup = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult & { user: UserEvent } => {
  return {
    user: userEvent.setup(),
    ...render(ui, options),
  };
};
