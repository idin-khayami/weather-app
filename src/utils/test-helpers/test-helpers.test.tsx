import { screen } from '@testing-library/react';

import { renderWithUserSetup } from '@/utils/test-helpers/test-helpers';

describe('Utils: test-helpers', () => {
  it('renderWithUserSetup should have set up userEvent correctly', async () => {
    const mockFn = jest.fn();

    const { user } = renderWithUserSetup(
      <button onClick={mockFn} data-testid="sampleButton">
        Hello World!
      </button>,
    );

    const element = screen.getByRole('button');

    expect(user).toBeDefined();
    expect(user).toHaveProperty('click');

    expect(mockFn).toHaveBeenCalledTimes(0);
    await user.click(element);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('renderWithUserSetup should have rendered component correctly', () => {
    renderWithUserSetup(<div data-testid="sampleDiv">Hello StuDocu</div>);

    const div = screen.getByTestId('sampleDiv');

    expect(div).toBeDefined();
    expect(div).toBeInTheDocument();
    expect(div).toMatchSnapshot();
  });
});
