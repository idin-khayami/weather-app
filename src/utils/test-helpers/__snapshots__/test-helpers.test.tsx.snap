// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils: test-helpers renderWithUserSetup should have rendered component correctly 1`] = `
<div
  data-testid="sampleDiv"
>
  Hello StuDocu
</div>
`;
